vyper/semantics/types/function.py

```diff
@@ -330,7 +330,22 @@ def from_vyi(cls, funcdef: vy_ast.FunctionDef) -> "ContractFunctionT":
         function_visibility, state_mutability, nonreentrant = _parse_decorators(funcdef)
 
         if nonreentrant:
-            raise FunctionDeclarationException("`@nonreentrant` not allowed in interfaces", funcdef)
+            decorator = next(d for d in funcdef.decorator_list if d.id == "nonreentrant")
```

Comment:
maybe `_parse_decorators` should actually return more structured output, so we don't need to do these searches (which is duplicating code with `_parse_decorators`)

Reply:
i think the current implementation is fairly simple and easy to refactor in the future. i favor avoiding further changes for this lightweight release

Reply:
add a comment then -- something along the lines of
```suggestion
            # TODO: refactor so parse_decorators returns the AST location
            decorator = next(d for d in funcdef.decorator_list if d.id == "nonreentrant")
```